modules/packages/Python.chpl

```diff
@@ -1186,7 +1204,32 @@ module Python {
       return arr;
     }
 
-    // TODO: convert python dict to chapel map
+    /*
+      Converts a python dictionary to a chapel map.  Steals a reference to obj.
+    */
+    @chpldoc.nodoc
+    proc fromDict(type T, obj: PyObjectPtr): T throws
+      where isSubtype(T, Map.map) {
+      var m = new T();
+      type keyType = m.keyType;
+      type valType = m.valType;
+
+      var keys = PyDict_Keys(obj);
+      defer Py_DECREF(keys);
+      this.checkException();
+      for i in 0..<PyList_Size(keys) {
+        var key = PyList_GetItem(keys, i);
+        this.checkException();
+        var val = PyDict_GetItem(obj, key);
+        this.checkException();
+
+        var keyVal = this.fromPython(keyType, key);
+        m.add(keyVal, this.fromPython(valType, val));
```

Comment:
fromPython steals a reference, but both PyList_GetItem and PyDict_GetItem return a borrowed reference. You should do an Py_INCREF before calling `fromPython`